### PR TITLE
feat: (1) RBS for models

### PR DESCRIPTION
**Pull request by tanmoysrt**
#### Describe the changes you have made in this PR -
- Fixed association for resolved `rbs_rails` issue
- Updated rbs collection
- Created rbs rake task
- Generated `rbs` files for `models` using `rbs_rails`
- Modified `rbs` files for `models` according to codebase
- Configure steep to 
   - enable checks for models 
   - suppress warning
   - suppress NoMethod Errors

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

**Comment by tanmoysrt** [comment]
This PR is too large. Can't split in smaller PR as maximum portion of the rbs has been generated by `rbs_rails` & it has  no command to generate single model's rbs at a time.

In the rbs files, the required function of codebase has been written at the end of the file. You can spot that portion by going to end and the portion after `ActiveRecord_Associations_CollectionProxy`

cc @vedant-jain03 @smritigarg @tachyons

**Comment by smritigarg** [comment]
> This PR is too large. Can't split in smaller PR as maximum portion of the rbs has been generated by `rbs_rails` & it has no command to generate single model's rbs at a time.
> 
> In the rbs files, the required function of codebase has been written at the end of each file. You can spot that portion by going to end of the file and the portion after `ActiveRecord_Associations_CollectionProxy`
> 
> Secondly, I have kept some functions as untyped and added `TODO` at the end of line, because for those classes `rbs` is still not written
> 
> cc @vedant-jain03 @smritigarg @tachyons

Thats alright I think for the current case, we need to just review the files which are not auto-generated.

**Review by vedant-jain03 — APPROVED** [review]
we can merge this, cc @tachyons

File: app/models/project.rb [hunk 1]
```diff
@@ -96,7 +95,8 @@ def toggle_star(user)
 
   def fork(user)
     forked_project = dup
-    forked_project.build_project_datum.data = project_datum&.data
+    forked_project.build_project_datum(data: project_datum&.data)
+    # forked_project.build_project_datum.data = project_datum&.data
```
Comment (smritigarg) [inline app/models/project.rb:99]: Shall we remove this commented line of code?

File: app/models/grade.rb [hunk 2]
```diff
@@ -37,7 +37,7 @@ def self.to_csv(assignment_id)
       attributes = %w[email name grade remarks]
       group_members = User.joins(group_members: :assignments)
                           .where(group_members: { assignments: { id: assignment_id } })
-      submissions = Project.where(assignment_id: assignment_id)&.includes(:grade, :author)
+      submissions = Project.where(assignment_id: assignment_id).includes(:grade, :author)
```
Comment (smritigarg) [inline app/models/grade.rb:40]: **Suggestion** Not sure if removing & is a good idea. If this does not concern the current set of changes then better not do the same

**Pull request closed without merging**